src/diem/jsonrpc/client.py

```diff
@@ -413,6 +413,23 @@ def get_account_state_with_proof(
         params = [address, version, ledger_version]
         return self.execute("get_account_state_with_proof", params, _parse_obj(lambda: rpc.AccountStateWithProof()))
 
+    def get_diem_id_domain_map(self, batch_size: int = 100) -> typing.Dict[str, str]:
+        domain_map = {}
+        event_index = 0
+        tc_account = self.must_get_account(utils.account_address(utils.TREASURY_ADDRESS))
+        event_stream_key = tc_account.role.diem_id_domain_events_key
+        while True:
+            events = self.get_events(event_stream_key, event_index, batch_size)
+            for event in events:
+                if event.data.removed:
+                    del domain_map[event.data.domain]
+                else:
+                    domain_map[event.data.domain] = event.data.address
+            if len(events) < batch_size:
+                break
+            event_index += batch_size
+        return domain_map
```

Comment:
Create a test in tests/test_testnet.py:

* create 2 onchain accounts with different domain
* remove one domain from one account
* call get_diem_id_domain_map to get the map, and verify the result